### PR TITLE
feat(config): introduce configuration builder

### PR DESCRIPTION
**Pull request by Altahrim**
### Content
Introduce a configuration builder based on JSON files:
 - `major_versions.json` declare all Nextcloud major versions with end of life and minimal supported PHP version
 - `releases.json` list all releases with their attributes (signature, internal version number and deploy percentage)

The builder should be able to generate the `config.php`  we are used to.

### TODO
- [x] Add `README.md`
- [x] Add check to verify config.php is up-to-date with the JSON files
- [x] Check generated config.php file (pass tests)
- [ ] Rebase before merge to relaunch tests with latest versions :warning:

### CODE REVIEW

**Comment by joshtrichards** [comment]
Nice! 

Since you're already stuffing the branch eol date into the json, this would be an ideal opportunity for #1050.

This would let me do this: https://github.com/nextcloud/server/issues/45402#issuecomment-2119280932

File: Makefile [hunk 1]
```diff
@@ -0,0 +1,5 @@
+config/config.php: config/major_versions.json config/releases.json $(wildcard config/entreprise_releases.json)
```
Comment (kesselb) [inline Makefile:1]:
```suggestion
config/config.php: config/major_versions.json config/releases.json $(wildcard config/enterprise_releases.json)
```

**Review by kesselb — COMMENTED** [review]
Cool :+1:

File: Makefile (hunk 1 above)
Comment (Altahrim) [reply Makefile:1]: Well spotted :)

File: config/major_versions.json [hunk 2]
```diff
@@ -0,0 +1,89 @@
...
+    "27": {
+        "eol": "2024-06",
+        "minPHP": "8.0"
+    },
+    "26": {
+        "eol": "2024-03",
+        "minPHP": "8.0"
+    },
+    "25": {
+        "eol": "2023-10 ",
```
[the remark this answers is not in the document]
Comment (Altahrim) [reply config/major_versions.json:22]: I'll update! Thank you :)

**Comment by Altahrim** [comment]
Rebased for latest `config.php` and fixed code/tests

**Pull request merged**